src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParser.cs

```diff
@@ -8747,10 +8776,16 @@ private Task WriteUnterminatedSqlValue(object value, MetaType type, int actualLe
 
                 case TdsEnums.SQLUNIQUEID:
                     {
+                        Debug.Assert(actualLength == 16, "Invalid length for guid type in com+ object");
+#if netcoreapp
+                        Span<byte> b = stackalloc byte[16];
+                        SqlGuid sqlGuid = (SqlGuid)value;
+                        (sqlGuid.IsNull ? Guid.Empty : sqlGuid.Value).TryWriteBytes(b);
```

Comment:
That was why I changed it. Rather than clear the buffer I chose to always write but switch the input for no particular reason I can articulate. If your suggestion is considered clearer I can go with that.

Reply:
At a guess `(sqlGuid.IsNull ? Guid.Empty : sqlGuid.Value)` will create an additional 16 byte local on the stack to hold the result of the ternary. Could be wrong though

Reply:
The stack frame has 30 locals, 31 with the if version. Now I've looked over it a few more times though I've changed my mind and I'll go with your version and include a comment on the clear.

Reply:
Any scope for making `SqlGuid`'s backing array `internal` and using that?

Reply:
Not directly. `SqlGuid` is defined in `System.Data.Common` assembly so you can't touch the internals and I doubt anyone would support introducing an `InternalsVisibleTo`. 

However there's the lovely [SqlTypeWorkarounds.cs](https://github.com/dotnet/corefx/blob/master/src/System.Data.SqlClient/src/System/Data/SqlTypes/SqlTypeWorkarounds.cs) which uses an overlay trick to touch the internals, I looked at using this and decided I'd rather not perpetuate it's use without good reason.

Reply:
Thinking about this more I could use the type workaround to avoid the byte array allocation in ToByteArray() in netstandard builds as well. Much as I don't like the overlay approach it has been in there a decade and does work. What do you think @saurabh500?

Reply:
> I would prefer not to use the overlay approach for newer optimizations.

Good enough for me, I'd made the same decision but thought I'd mention the option.

Reply:
Sounds good. Leaving this PR link here https://github.com/dotnet/corefx/pull/12426 where Overlays were introduced and the reasoning behind them.